sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1666,7 +1667,14 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
                   // The update value can access columns from both target and source tables.
                   resolveAssignments(assignments, m, MergeResolvePolicy.BOTH))
               case UpdateStarAction(updateCondition) =>
-                val assignments = targetTable.output.map { attr =>
+                val attrs = if (m.withSchemaEvolution) {
+                  // For UPDATE *, the column must be from source table, even with schema evolution.
```

Comment:
I thought `UPDATE *` should assign value to each column in the target table. What's the semantic of some columns are omitted?

Reply:
yea its a bit of a new feature for Spark MERGE INTO, Delta Lake MERGE INTO support this case and makes the new value null.

https://docs.delta.io/latest/delta-update.html#automatic-schema-evolution 

Actually I am adding this support for top level field assignment here, we still dont have the support for struct type where source has less struct fields as I mention in the pr description.

Do you think I should do it in a separate pr?